src/core/prometheus.cc

```diff
@@ -800,19 +800,19 @@ class metrics_handler : public httpd::handler_base  {
 
     future<std::unique_ptr<http::reply>> handle(const sstring& path,
         std::unique_ptr<http::request> req, std::unique_ptr<http::reply> rep) override {
-        auto text = is_accept_text(req->get_header("Accept"));
+        auto is_text_format = !_ctx.allow_protobuf || is_accept_text(req->get_header("Accept"));
```

Comment:
I don't understand the logic. Suppose the header only accepts protobuf. We should respond with an error of protobuf is not configured, instead we respond with text.

Reply:
We are keeping the current behavior, right now, Prometheus does not check the header and will return text format regardless of what the client asks.
I believe this is an ok behavior.

If we want to change the current behavior, it should be done with an explicit issue and a warning that theoretically it could be a breaking change.

Reply:
ok, I see is_accept_text() is really is_not_accept_protobuf().

Reply:
@avikivity I've updated the title and cover